hardlydifficult/LD40
# WIP on spells system

- new PlayerSpellsManager to hold the state and add events on changes
  -> to be added in Controllers (exported in Controllers.prefab)
- new SpellFloating component that let an object float when a spell is active
  -> to be added on the main camera (exported in Main Camera.prefab)
  -> to be added on cauldrons (exported in TheBoard_UpdatedSpells.prefab, but CONFLICT with other changes on the board)
- HitDetector now fires a given spell
  -> to be added on cauldrons (exported in TheBoard_UpdatedSpells.prefab, but CONFLICT with other changes on the board)

To test:
- integrate Controllers.prefab and Main Camera.prefab into the Scenes
- Play, then in the Controllers PlayerSpellsManager, enable the checkbox on the first spell (butterbeer) and you should see the camera moving
- if TheBoard_UpdatedSpells.prefab is integrated (for testing), there are editorHit checkbox on each HitDetector to enable the spell from this cauldron (and the spell 3 'animatio' makes the cauldron float)

I can re-merge the board part into the new one after the merge of this PR.
